### ISSUE TITLE
Headers should be handled case insensitive to prevent dupes and failed overrides

### ISSUE THREAD

**Issue by icyrizard** [zapier/zapier-platform-cli#111]
The documentation states that we can use `'Content-Type'` as a header parameter. I've discovered that this option is case sensitive and is ignored if capital letters are used. I've ran tests with ngrok so I could inspect what Zapier actually sends when changing this from `Content-Type` -> `content-type`.

It took me a lot more hours to establish an OAuth2 connection, so fixing this could also help others in their development. In the end, the content-type will determine how the data is interpreted by the party that I'm trying to connect to.

The simplest fix would be to update the docs, but the real fix would be to not ignore the Content-Type, because that is i.m.o correct (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)

## To be elaborate:

**Wrong:**

~~~javascript
 const getAccessToken = (z, bundle) => {
  const promise = z.request({
    method: 'POST',
    url: '{{bundle.inputData.token_url}}',
    body: {
        code: bundle.inputData.code,
        redirect_uri: bundle.inputData.redirect_uri,
        client_id: '{{process.env.CLIENT_ID}}',
        client_secret: '{{process.env.CLIENT_SECRET}}',
        grant_type: 'authorization_code'
    },
    headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
    }
  });

...
~~~

![screenshot at jul 25 17-55-29](https://user-images.githubusercontent.com/681775/28581508-aa40ab66-7162-11e7-8d0f-6acb21f0b68c.png)

**Correct:**

~~~javascript
 const getAccessToken = (z, bundle) => {
  const promise = z.request({
    method: 'POST',
    url: '{{bundle.inputData.token_url}}',
    body: {
        code: bundle.inputData.code,
        redirect_uri: bundle.inputData.redirect_uri,
        client_id: '{{process.env.CLIENT_ID}}',
        client_secret: '{{process.env.CLIENT_SECRET}}',
        grant_type: 'authorization_code'
    },
    headers: {
        'content-type': 'application/x-www-form-urlencoded'
    }
  });

....

~~~

![screenshot at jul 25 17-55-17](https://user-images.githubusercontent.com/681775/28581483-9876e8c8-7162-11e7-8721-a15bd4486db7.png)

**Comment by bryanhelmig** [comment]
Thanks for the report @icyrizard -- we'll get that fixed up!

**Comment by icyrizard** [comment]
Awesome stuff! :)

**Comment by FokkeZB** [comment]
Just had another report of this and although the above screenshots don't show it, it seems like request doesn't do a case insensitive dedupe of the headers.

We insert them in lower case:
https://github.com/zapier/zapier-platform-core/blob/c06fb327155663d5f2cb6d44562112b18cc2845a/src/http-middlewares/before/prepare-request.js#L23-L29

So if the developer adds the same headers in different casing you will end up with duplicate headers in the request:

https://gist.github.com/codekirei/bdb544098237e62f7f4580599d21feb8#file-400-http-L12-L14

We need to dedupe the headers case insensitive and make sure the user can override https://github.com/zapier/zapier-platform-core/blob/c06fb327155663d5f2cb6d44562112b18cc2845a/src/http-middlewares/before/prepare-request.js#L23-L29

**Comment by FokkeZB** [comment]
Perhaps we can use https://www.npmjs.com/package/caseless

**Comment by BrunoBernardino** [comment]
This should be fixed in `4.0.0`, please let us know if otherwise!
